delta/plugins/graph-analytics/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/graph/analytics/indexing/GraphAnalyticsStream.scala

```diff
@@ -91,10 +91,15 @@ object GraphAnalyticsStream {
             UpdateByQuery(s.id, s.types)
           }
         case Resources.entityType =>
-          Task.fromEither(ResourceState.serializer.codec.decodeJson(json)).flatMap { s =>
-            JsonLdDocument.fromExpanded(s.expanded, findRelationships(project, xas, relationshipBatch)).map { d =>
-              Index(s.project, s.id, s.rev, s.types, s.createdAt, s.createdBy, s.updatedAt, s.updatedBy, d)
-            }
+          Task.fromEither(ResourceState.serializer.codec.decodeJson(json)).flatMap {
+            case s if s.deprecated =>
+              Task.pure(
+                Index.deprecated(s.project, s.id, s.rev, s.types, s.createdAt, s.createdBy, s.updatedAt, s.updatedBy)
+              )
+            case s                 =>
+              JsonLdDocument.fromExpanded(s.expanded, findRelationships(project, xas, relationshipBatch)).map { d =>
+                Index.active(s.project, s.id, s.rev, s.types, s.createdAt, s.createdBy, s.updatedAt, s.updatedBy, d)
+              }
```

Comment:
Could potentially be more readable if `index.active`, `index.deprecated` took in a state

Reply:
I think it is a good idea but that would be a bit outside of this PR.
We should introduce a `Metadata` class that can group all these properties to make them easier to use.
It could be applied there and in other places.